alicloud/resource_alicloud_slb_acl.go

```diff
@@ -0,0 +1,316 @@
+			return err
+		}
+	}
+
+	return nil
+}
+
+func slbAddAccessControlListEntryPerTime(client *slb.Client, list []interface{}, aclId string) error {
+	req := slb.CreateAddAccessControlListEntryRequest()
+	req.AclId = aclId
+	b, _ := json.Marshal(list)
+	req.AclEntrys = string(b)
+	if _, err := client.AddAccessControlListEntry(req); err != nil {
+		return fmt.Errorf("AddAccessControlListEntry got an error: %#v", err)
+	}
+
+	return nil
+}
+
+func slbAddAccessControlListEntry(client *slb.Client, list []interface{}, aclId string) error {
+	num := len(list)
+
+	if num <= 0 {
+		return nil
+	}
+
+	t := (num + max_num_per_time - 1) / max_num_per_time
+	for i := 0; i < t; i++ {
+		start := i * max_num_per_time
+		end := (i + 1) * max_num_per_time
+
+		if end > num {
+			end = num
+		}
+		slice := list[start:end]
+		if err := slbAddAccessControlListEntryPerTime(client, slice, aclId); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Flattens an array of slb.AclEntry into a []map[string]string
+func flattenSlbAclEntryMappings(list []slb.AclEntry) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, len(list))
+
+	for _, i := range list {
+		l := map[string]interface{}{
+			"entry":   i.AclEntryIP,
+			"comment": i.AclEntryComment,
+		}
+		result = append(result, l)
+	}
+
+	return result
+}
+
+// Flattens an array of slb.AclEntry into a []map[string]string
+func flattenSlbRelatedListeneryMappings(list []slb.RelatedListener) []map[string]interface{} {
+	result := make([]map[string]interface{}, 0, len(list))
+
+	for _, i := range list {
+		l := map[string]interface{}{
+			"load_balancer_id": i.LoadBalancerId,
+			"protocol":         i.Protocol,
+			"frontend_port":    i.ListenerPort,
+			"acl_type":         i.AclType,
+		}
+		result = append(result, l)
+	}
+
+	return result
+}
+
+func resourceAlicloudSlbAcl() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudSlbAclCreate,
+		Read:   resourceAlicloudSlbAclRead,
+		Update: resourceAlicloudSlbAclUpdate,
+		Delete: resourceAlicloudSlbAclDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+				Default:  "tf-slb-acl",
+			},
+
+			"ip_version": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "ipv4",
+			},
+			"entrys": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"entry": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"comment": &schema.Schema{
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+				MaxItems: 300,
+				MinItems: 0,
+			},
+			"related_listeners": &schema.Schema{
+				Type:     schema.TypeList,
```

Comment:
I think there is no need to show `related_listeners ` in this resource, and it can be shown in the data source.

Reply:
`related_listeners` the dynamic computed attribute of  `alicould_slb_acl`, may be useful  to its  related listeners.